### PR TITLE
Implemetation of lab 02

### PR DESCRIPTION
**Pull request by AhmedEldemoksy**

### CODE REVIEW

File: 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/Sql_Statments.sql [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                        constraint place_fk1 foreign key (pp_id) references places (place_id)
+);
+create table visits (user_id int, 
+					place_id int, 
+                    tim time,
+                    primary key (user_id, place_id),
+                    constraint user_fk4 foreign key (user_id) references users (user_id),
+					constraint place_fk2 foreign key (place_id) references places (place_id)
+);
+show tables;
```
Comment (helghareeb) [inline 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/Sql_Statments.sql:49]: Remove `show tables` at line 49

File: 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/Sql_Statments.sql [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                        primary key (keyword_id),
+                        constraint category_fk2 foreign key (cid) references category (category_id)
+);
+create table keymedia (keyword_id int,
+						med_id int,
+                        primary key (keyword_id, med_id),
+                        constraint keyword_fk1 foreign key (keyword_id) references keyword (keyword_id),
+                        constraint media_fk6 foreign key (med_id) references mediaitems (med_id)
+);
+show tables;
```
Comment (helghareeb) [inline 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/Sql_Statments.sql:92]: Remove `show tables` at line 92

File: 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/Sql_Statments.sql [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
```
Comment (helghareeb) [inline 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/Sql_Statments.sql:93]: Remove blank lines at the end of the files

File: 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/hostoricalplaces.sql [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+
+DROP TABLE IF EXISTS `audiofiles`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!40101 SET character_set_client = utf8 */;
+CREATE TABLE `audiofiles` (
+  `med_id` int(11) NOT NULL,
+  `length` int(255) DEFAULT NULL,
+  PRIMARY KEY (`med_id`),
+  CONSTRAINT `media_fk4` FOREIGN KEY (`med_id`) REFERENCES `mediaitems` (`med_id`)
+) ENGINE=InnoDB DEFAULT CHARSET=latin1;
```
Comment (helghareeb) [inline 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/hostoricalplaces.sql:30]: Change CHARSET to `utf8`

File: 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/hostoricalplaces.sql [hunk 5]
```diff
@@ -0,0 +1,350 @@
...
+/*!40101 SET character_set_client = utf8 */;
+CREATE TABLE `category` (
+  `category_id` int(11) NOT NULL,
+  `title` varchar(50) DEFAULT NULL,
+  `description` varchar(255) DEFAULT NULL,
+  `pcid` int(11) DEFAULT NULL,
+  PRIMARY KEY (`category_id`),
+  KEY `category_fk1` (`pcid`),
+  CONSTRAINT `category_fk1` FOREIGN KEY (`pcid`) REFERENCES `category` (`category_id`)
+) ENGINE=InnoDB DEFAULT CHARSET=latin1;
```
Comment (helghareeb) [inline 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/hostoricalplaces.sql:57]: Change CHARSET to `utf8`

File: 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/hostoricalplaces.sql [hunk 6]
```diff
@@ -0,0 +1,350 @@
...
+DROP TABLE IF EXISTS `imagefiles`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!40101 SET character_set_client = utf8 */;
+CREATE TABLE `imagefiles` (
+  `med_id` int(11) NOT NULL,
+  `width` int(255) DEFAULT NULL,
+  `height` int(255) DEFAULT NULL,
+  PRIMARY KEY (`med_id`),
+  CONSTRAINT `media_fk5` FOREIGN KEY (`med_id`) REFERENCES `mediaitems` (`med_id`)
+) ENGINE=InnoDB DEFAULT CHARSET=latin1;
```
Comment (helghareeb) [inline 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/hostoricalplaces.sql:82]: Change CHARSET to `utf8`

File: 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/hostoricalplaces.sql [hunk 7]
```diff
@@ -0,0 +1,350 @@
...
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!40101 SET character_set_client = utf8 */;
+CREATE TABLE `keymedia` (
+  `keyword_id` int(11) NOT NULL,
+  `med_id` int(11) NOT NULL,
+  PRIMARY KEY (`keyword_id`,`med_id`),
+  KEY `media_fk6` (`med_id`),
+  CONSTRAINT `keyword_fk1` FOREIGN KEY (`keyword_id`) REFERENCES `keyword` (`keyword_id`),
+  CONSTRAINT `media_fk6` FOREIGN KEY (`med_id`) REFERENCES `mediaitems` (`med_id`)
+) ENGINE=InnoDB DEFAULT CHARSET=latin1;
```
Comment (helghareeb) [inline 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/hostoricalplaces.sql:108]: Change CHARSET to `utf8`

File: 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/hostoricalplaces.sql [hunk 8]
```diff
@@ -0,0 +1,350 @@
...
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!40101 SET character_set_client = utf8 */;
+CREATE TABLE `keyword` (
+  `keyword_id` int(11) NOT NULL,
+  `term` varchar(30) DEFAULT NULL,
+  `cid` int(11) DEFAULT NULL,
+  PRIMARY KEY (`keyword_id`),
+  KEY `category_fk2` (`cid`),
+  CONSTRAINT `category_fk2` FOREIGN KEY (`cid`) REFERENCES `category` (`category_id`)
+) ENGINE=InnoDB DEFAULT CHARSET=latin1;
```
Comment (helghareeb) [inline 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/hostoricalplaces.sql:134]: Change CHARSET to `utf8`

File: 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/hostoricalplaces.sql [hunk 9]
```diff
@@ -0,0 +1,350 @@
...
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!40101 SET character_set_client = utf8 */;
+CREATE TABLE `mediaitems` (
+  `med_id` int(11) NOT NULL,
+  `title` varchar(50) DEFAULT NULL,
+  `description` varchar(20) DEFAULT NULL,
+  `URL` varchar(80) DEFAULT NULL,
+  `size` varchar(50) DEFAULT NULL,
+  PRIMARY KEY (`med_id`)
+) ENGINE=InnoDB DEFAULT CHARSET=latin1;
```
Comment (helghareeb) [inline 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/hostoricalplaces.sql:160]: Change CHARSET to `utf8`

File: 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/hostoricalplaces.sql [hunk 10]
```diff
@@ -0,0 +1,350 @@
...
+  `place_name` varchar(50) DEFAULT NULL,
+  `description` varchar(50) DEFAULT NULL,
+  `user_creator` int(11) DEFAULT NULL,
+  `pp_id` int(11) DEFAULT NULL,
+  PRIMARY KEY (`place_id`),
+  KEY `user_fk3` (`user_creator`),
+  KEY `place_fk1` (`pp_id`),
+  CONSTRAINT `place_fk1` FOREIGN KEY (`pp_id`) REFERENCES `places` (`place_id`),
+  CONSTRAINT `user_fk3` FOREIGN KEY (`user_creator`) REFERENCES `users` (`user_id`)
+) ENGINE=InnoDB DEFAULT CHARSET=latin1;
```
Comment (helghareeb) [inline 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/hostoricalplaces.sql:190]: Change CHARSET to `utf8`

File: 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/hostoricalplaces.sql [hunk 11]
```diff
@@ -0,0 +1,350 @@
...
+  `place_id` int(11) NOT NULL,
+  `med_id` int(11) NOT NULL,
+  `tim` time NOT NULL,
+  PRIMARY KEY (`user_id`,`place_id`,`med_id`,`tim`),
+  KEY `place_fk3` (`place_id`),
+  KEY `media_fk3` (`med_id`),
+  CONSTRAINT `media_fk3` FOREIGN KEY (`med_id`) REFERENCES `mediaitems` (`med_id`),
+  CONSTRAINT `place_fk3` FOREIGN KEY (`place_id`) REFERENCES `places` (`place_id`),
+  CONSTRAINT `user_fk5` FOREIGN KEY (`user_id`) REFERENCES `users` (`user_id`)
+) ENGINE=InnoDB DEFAULT CHARSET=latin1;
```
Comment (helghareeb) [inline 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/hostoricalplaces.sql:220]: Change CHARSET to `utf8`

File: 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/hostoricalplaces.sql [hunk 12]
```diff
@@ -0,0 +1,350 @@
...
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!40101 SET character_set_client = utf8 */;
+CREATE TABLE `users` (
+  `user_id` int(11) NOT NULL,
+  `user_name` varchar(50) DEFAULT NULL,
+  `email` varchar(20) DEFAULT NULL,
+  `imageURL` varchar(80) DEFAULT NULL,
+  `interests` varchar(80) DEFAULT NULL,
+  PRIMARY KEY (`user_id`)
+) ENGINE=InnoDB DEFAULT CHARSET=latin1;
```
Comment (helghareeb) [inline 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/hostoricalplaces.sql:246]: Change CHARSET to `utf8`

File: 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/hostoricalplaces.sql [hunk 13]
```diff
@@ -0,0 +1,350 @@
...
+CREATE TABLE `views` (
+  `med_id` int(11) NOT NULL,
+  `user_id` int(11) NOT NULL,
+  `tim` time NOT NULL,
+  `durnation` varchar(50) DEFAULT NULL,
+  PRIMARY KEY (`med_id`,`user_id`,`tim`),
+  KEY `users_fk1` (`user_id`),
+  CONSTRAINT `media_fk1` FOREIGN KEY (`med_id`) REFERENCES `mediaitems` (`med_id`),
+  CONSTRAINT `users_fk1` FOREIGN KEY (`user_id`) REFERENCES `users` (`user_id`)
+) ENGINE=InnoDB DEFAULT CHARSET=latin1;
```
Comment (helghareeb) [inline 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/hostoricalplaces.sql:274]: Change CHARSET to `utf8`

File: 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/hostoricalplaces.sql [hunk 14]
```diff
@@ -0,0 +1,350 @@
...
+/*!40101 SET character_set_client = utf8 */;
+CREATE TABLE `visits` (
+  `user_id` int(11) NOT NULL,
+  `place_id` int(11) NOT NULL,
+  `tim` time DEFAULT NULL,
+  PRIMARY KEY (`user_id`,`place_id`),
+  KEY `place_fk2` (`place_id`),
+  CONSTRAINT `place_fk2` FOREIGN KEY (`place_id`) REFERENCES `places` (`place_id`),
+  CONSTRAINT `user_fk4` FOREIGN KEY (`user_id`) REFERENCES `users` (`user_id`)
+) ENGINE=InnoDB DEFAULT CHARSET=latin1;
```
Comment (helghareeb) [inline 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/hostoricalplaces.sql:301]: Change CHARSET to `utf8`

File: 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/hostoricalplaces.sql [hunk 15]
```diff
@@ -0,0 +1,350 @@
...
+CREATE TABLE `votes` (
+  `med_id` int(11) NOT NULL,
+  `user_id` int(11) NOT NULL,
+  `rating` varchar(50) DEFAULT NULL,
+  `tim` time DEFAULT NULL,
+  PRIMARY KEY (`med_id`,`user_id`),
+  KEY `users_fk2` (`user_id`),
+  CONSTRAINT `media_fk2` FOREIGN KEY (`med_id`) REFERENCES `mediaitems` (`med_id`),
+  CONSTRAINT `users_fk2` FOREIGN KEY (`user_id`) REFERENCES `users` (`user_id`)
+) ENGINE=InnoDB DEFAULT CHARSET=latin1;
```
Comment (helghareeb) [inline 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/hostoricalplaces.sql:329]: Change CHARSET to `utf8`

**Review by helghareeb — CHANGES_REQUESTED** [review]

File: 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/Sql_Statments.sql (hunk 1 above)
Comment (AhmedEldemoksy) [reply 03-04-DBFundamentals_MySQL/Day-04/Lab-02-Solution/Implementation/Sql_Statments.sql:49]: All is done

**Comment by AhmedEldemoksy** [comment]
All is done

**Pull request closed without merging**